### ISSUE TITLE
Roslyn doesn't report errors for unknown attributes in local functions' type parameters lists

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#14640]
Roslyn doesn't report any errors for this code sample (it should report unresolved attribute)

``` cs
public class C {
    public static void Main()
    {
        void Local<[A] T>(){}
        Local<int>();
     }
}
```

attempts to compile this code from VS15 Preview 5 ends with following error (which is not very helpful)

```
"csc.exe" exited with code -532462766.
```

currently this test fails 

``` cs
        [Fact]
        public void UnknownAttribute()
        {
            var src = @"
public class C {
    public static void Main()
    {
        void Local<[A] T>(){}
        Local<int>();
     }
}";
          VerifyDiagnostics(src,
            // (5,21): error CS0246: The type or namespace name 'AAttribute' could not be found (are you missing a using directive or an assembly reference?)
            //         void Local<[A] T>(){}
            Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "A").WithArguments("AAttribute").WithLocation(5, 21),
            // (5,21): error CS0246: The type or namespace name 'A' could not be found (are you missing a using directive or an assembly reference?)
            //         void Local<[A] T>(){}
            Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "A").WithArguments("A").WithLocation(5, 21));
        }
```

with following output:

```
Expected:
    Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "A").WithArguments("AAttribute"),
    Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "A").WithArguments("A")
Actual:
```

**Comment by jcouv** [comment]
Tracked by VSO https://devdiv.visualstudio.com/DevDiv/_workitems?id=369810&_a=edit
